Sources/AblyAssetTrackingInternal/WorkerQueue/Worker.swift

```diff
@@ -0,0 +1,64 @@
+import Foundation
+import AblyAssetTrackingCore
+
+/// A [Worker] interface represents workers which execute work inside [WorkerQueue].
+/// PropertiesType: - the type of properties used by this worker as both input and output.
+/// WorkerSpecificationType - the type of specification used to post another worker back to the queue.
+public protocol Worker<PropertiesType, WorkerSpecificationType>: AnyObject {
+    associatedtype PropertiesType
+    associatedtype WorkerSpecificationType
+    
+    var completion: ResultHandler<Void> { get }
+    var workerTypeDescription: String { get }
```

Comment:
I definitely should have documented this one, and I will if we decide to stick with this solution - the intention is to log unexpected errors from async work in WorkerQueue using this var:
```
self.logHandler?.error(message: "Unexpected error thrown from the asynchronous work of \(worker.workerTypeDescription)", error: error)
```
I couldn't find a way to use:
```
self.logHandler?.error(message: "Unexpected error thrown from the asynchronous work of \(worker.self)", error: error)
```
or anything like that, since `worker` here is of type `Worker<PropertiesType, WorkerSpecificationType>`, and code doesn't compile (Xcode allows us to access `self` on objects like that only starting with iOS 16).

So the intended workaround here is for each worker to assign a value to this property on it's own (which is not ideal of course).

Reply:
Maybe I've missed something, but could we not use `\(type(of: worker))` in the log message?

Reply:
Hmm, I was sure I tried using and had problems as described above, but it seems to be working now 🤷 860918f4213211b125b4b7150d084d69e175e086